### PR TITLE
stm32l1: initial port for the nucleo-l1 board and stm32l1 cpu

### PR DESCRIPTION
**Pull request by thomaseichinger**
Known issues:
* ~~shell input currently not working~~

### CODE REVIEW

File: boards/nucleo-l1/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,56 @@
+## the cpu to build for
+export CPU = stm32l1
+export CPU_MODEL = stm32l152ret6
+
+#define the default port depending on the host OS
+OS := $(shell uname)
+ifeq ($(OS),Linux)
+  PORT ?= /dev/ttyUSB0
```
Comment (haukepetersen) [inline boards/nucleo-l1/Makefile.include]: change this to `/dev/ttyACM0`, and the board is able to communicate out-of-the-box using the integrated USB-UART adapter without any external hw...

File: boards/nucleo-l1/include/board.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ */
+void board_init(void);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* BOARD_H_ */
+/** @} */
+/** @} */
```
Comment (haukepetersen) [inline boards/nucleo-l1/include/board.h]: seems like there is on `/*@} */` too much?!

File: boards/nucleo-l1/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+/**
+ * @brief SPI configuration
+ * @{
+ */
+#define SPI_NUMOF           (1U)
+#define SPI_0_EN            1
+
+/* SPI 0 device configuration */
+#define SPI_0_DEV               SPI1
```
Comment (haukepetersen) [inline boards/nucleo-l1/include/periph_conf.h]: looks like the indention is off by 2 spaces...

File: cpu/stm32l1/include/cpu-conf.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ */
+typedef enum {
+    HARD_FAULT,
+    WATCHDOG,
+    BUS_FAULT,
+    USAGE_FAULT,
+    DUMMY_HANDLER
+} panic_t;
+
+void cpu_clock_scale(uint32_t source, uint32_t target, uint32_t *prescale);
```
Comment (haukepetersen) [inline cpu/stm32l1/include/cpu-conf.h]: defined but never used/implemented?

File: cpu/stm32l1/periph/cpuid.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+#include <string.h>
+#include "cpu-conf.h"
+
+#include "periph/cpuid.h"
+
+void cpuid_get(void *id)
+{
+    memcpy(id, (void *)(0x1ff800d0), CPUID_ID_LEN);
```
Comment (haukepetersen) [inline cpu/stm32l1/periph/cpuid.c]: magic number (`0x1ff800d0`), how about putting in a define for readability reasons?

File: boards/nucleo-l1/Makefile.features [hunk 6]
```diff
@@ -0,0 +1 @@
+FEATURES_PROVIDED += periph_gpio periph_uart periph_spi periph_i2c periph_cpuid
```
Comment (haukepetersen) [inline boards/nucleo-l1/Makefile.features]: add also `cpp` as it works fine :-)

**Comment by haukepetersen** [comment]
ok, just found some minor things, will ack once these are fixed. Good job!

**Comment by haukepetersen** [comment]
and don't forget to squash...

File: boards/nucleo-l1/include/board.h (hunk 2 above)
Comment (thomaseichinger) [inline boards/nucleo-l1/include/board.h]: I added this in 1548421fbec787ad1d7a7838b95d8258df591f4c to fix doxygen warnings.

**Comment by thomaseichinger** [comment]
addressed @haukepetersen's comments, rebased and squashed.

File: boards/nucleo-l1/Makefile.include (hunk 1 above)
Comment (OlegHahm) [inline boards/nucleo-l1/Makefile.include]: Didn't work on my PC.

**Comment by thomaseichinger** [comment]
I changed the wiki in respect of the UART mystery. Travis passed.

**Comment by haukepetersen** [comment]
Looking good. ACK and go.

**Pull request merged**